pkg/api/filters/authenticator/authenticator.go

```diff
@@ -0,0 +1,8 @@
+package authenticator
+
+import (
+	"github.com/Azure/open-service-broker-azure/pkg/api/filters"
+)
+
+// Authenticator is an alias of filters.Filter that represents an authenticator
+type Authenticator filters.Filter
```

Comment:
I haven't refactored the broker interface yet. It still takes in the Authenticator type instead of a filter chain. I wasn't sure if I wanted to contain that in the broker or expose that so the builder of the broker (i.e. cmd/broker/broker.go) should determine the configuration. I'm leaning that way, since we construct other things there. (i.e. the authenticator, redis, etc). Maybe move the filter chain construction here.

Reply:
Yes. There are most assuredly spots where we're _not_ doing this, but I think we should generally favor treating the `cmd/broker` code as a factory of sorts. It will instantiate all the right implementations of things (based on configuration) and use "constructor injection" to furnish lower-level code with its dependencies.